port : $(arg driver_port)
baud : 115200

# specify robot track width in meters
track_width: 1.0025 # +1 cm compared to aristos' 0.9925
# specify reduction ratio (70:1), if there is no put 1
reduction_ratio: 70
# specify reduction maximum linear velocity (m/s)
# max_vel: 0.95  # 60 m/min @3500rpm
max_vel: 0.42  # reduced speed for testing
# specify robot wheel circumference (perimeter) in meters 
wheel_circumference: 1.162389 # D:37cm
encoder_cpr: 1000 # 1000 pulses per revolution * 4, because of the quadratic mode  
encoder_min: -2147483648.0
encoder_max:  2147483648.0 
rate: 20 
enable_imu_yaw: $(arg use_imu)
imu_topic: $(arg imu_topic)
wrp_lim: 0.3
sub_to_abs: false
imu_discarded: 100
tf_publish: $(arg pub_odom_tf)
odom_topic: $(arg odom_topic)
tf_header_frame: $(arg prefix)odom
tf_child_frame: $(arg prefix)base_footprint
odom_frame: $(arg prefix)odom


channel_mode: dual
motor_type: skid_steering
# motor_1_type: skid_steering
# motor_2_type: skid_steering


# Frequency in Hz
frequency_list:
  - 100
  - 10
  - 25


# Queries order matches the order of the frequency list above 
# e.g query0 corresponds to the first element in the frequency list

query0:
# hall_count : ?BCR
# abs_hall_count : ?CB
# hall_speed : ?BS
#  status_flag : ?FS
# motor_amps2 : ?MA
# motor_command2 : ?FM
# hall_count2 : ?FM
# hall_speed2 : ?FS
# volts2 : ?V
# feedback2 : ?rmp 
# battery_amps2 : ?PHA
# power2 : ?sns          
# fault_flag2 : ?S  
# status_flag : ?FS
# firmware_id : ?FID  
 encoder_speed : ?S
 encoder_count_relative : ?CR
 encoder_count : ?C
   

query1:
 motor_amps : ?A 
 motor_command : ?M
 volts : ?V
 fault_flag : ?FF  
 sto_status : ?STT
 feedback : ?F 
 battery_amps : ?BA
 power : ?P 
 encoder_count : ?C

query2:
 motor_amps2 : ?MA
 hall_count : ?BCR
 abs_hall_count : ?CB